src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -268,3 +299,37 @@ function validate_connectivity(M, nodes, bus_lookup)
     end
     return connected
 end
+
+"""
+Computes the connected components of the network graph using network traversal.
+Returns a set of sets of buses, each set is a connected component
+"""
+# this function borrows heavily from "calc_connected_compoents" in PowerModels.jl
+function find_connected_components(sys::System)
+    buses = get_components(Bus, sys, x -> get_bustype(x) != BusTypes.ISOLATED)
+    branches = get_components(Branch, sys, get_available)
+
+    return find_connected_components(branches, buses)
+end
+
+function find_connected_components(branches, buses)
+    neighbors = Dict(i => [] for i in buses)
+    for line in branches
+        arc = get_arc(line)
+        if get_from(arc) ∈ buses && get_to(arc) ∈ buses
+            push!(neighbors[get_from(arc)], get_to(arc))
+            push!(neighbors[get_to(arc)], get_from(arc))
+        end
+    end
+
+    component_lookup = Dict(i => Set{Bus}([i]) for i in buses)
```

Comment:
I'm not sure that this is valid for mutable objects. Could you use bus number as the key? `IdDict{Bus}` is valid.